src/plugins/workspace/public/components/workspace_form/workspace_permission_setting_input.tsx

```diff
@@ -0,0 +1,128 @@
+/*
+ * Copyright OpenSearch Contributors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+import React, { useCallback, useMemo } from 'react';
+import {
+  EuiFlexGroup,
+  EuiComboBox,
+  EuiFlexItem,
+  EuiButtonIcon,
+  EuiButtonGroup,
+} from '@elastic/eui';
+import { WorkspacePermissionMode } from '../../../common/constants';
+import {
+  WorkspacePermissionItemType,
+  optionIdToWorkspacePermissionModesMap,
+  permissionModeOptions,
+} from './constants';
+import { getPermissionModeId } from './utils';
+
+export interface WorkspacePermissionSettingInputProps {
+  index: number;
+  deletable: boolean;
+  type: WorkspacePermissionItemType;
+  userId?: string;
+  group?: string;
+  modes?: WorkspacePermissionMode[];
+  onGroupOrUserIdChange: (
+    groupOrUserId:
+      | { type: WorkspacePermissionItemType.User; userId?: string }
+      | { type: WorkspacePermissionItemType.Group; group?: string },
+    index: number
+  ) => void;
+  onPermissionModesChange: (
+    WorkspacePermissionMode: WorkspacePermissionMode[],
+    index: number
+  ) => void;
+  onDelete: (index: number) => void;
+}
+
+export const WorkspacePermissionSettingInput = ({
+  index,
+  type,
+  userId,
+  group,
+  modes,
+  deletable,
+  onDelete,
+  onGroupOrUserIdChange,
+  onPermissionModesChange,
+}: WorkspacePermissionSettingInputProps) => {
+  const groupOrUserIdSelectedOptions = useMemo(
+    () => (group || userId ? [{ label: (group || userId) as string }] : []),
+    [group, userId]
+  );
+
+  const permissionModesSelectedId = useMemo(() => getPermissionModeId(modes ?? []), [modes]);
```

Comment:
do we need to handle if ground and userId both missing, and modes missing here?

Reply:
modes is a independent property, it's not rely on group and userId property. The `getPermissionModeId` will return `Read` for empty modes. And we are using  Read for default permission setting.